pkg/storage/unified/sql/service.go

```diff
@@ -258,3 +259,11 @@ func NewAuthenticatorWithFallback(cfg *setting.Cfg, reg prometheus.Registerer, t
 		return a.Authenticate(ctx)
 	}
 }
+
+func (s *service) stopping(err error) error {
```

Comment:
fyi, `err` passed to `stopping` method is result of `running` method. It might have been easier to filter out `context.Canceled` errors there.